test/libsolidity/SemanticTest.cpp

```diff
@@ -142,21 +170,40 @@ TestCase::TestResult SemanticTest::runTest(ostream& _stream, string const& _line
 	if (_compileViaYul)
 		AnsiColorized(_stream, _formatted, {BOLD, CYAN}) << _linePrefix << "Running via Yul:" << endl;
 
-	for (auto& test: m_tests)
+	for (TestFunctionCall& test: m_tests)
 		test.reset();
 
 	map<string, solidity::test::Address> libraries;
 
 	bool constructed = false;
 
-	for (auto& test: m_tests)
+	// Iterate through the test calls and set the previous call.
+	TestFunctionCall* previousCall{nullptr};
+	for (TestFunctionCall& test: m_tests)
+	{
+		test.setPreviousCall(previousCall);
+		test.setTestHooks(&m_testHooks);
+		previousCall = &test;
+	}
+
+	for (std::shared_ptr<TestHook>& hook: m_testHooks)
```

Comment:
Is `shared_ptr` instead of `unique_ptr` needed here? Also please use a const ref.

Reply:
True, I think `unique_ptr` should work.